lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -46,6 +46,24 @@ function dosomething_user_init() {
       dosomething_user_non_staff_redirect();
     }
   }
+
+  // If viewing a user/[northstar id]
+  if(arg(0) == 'user') {
+    if(!is_numeric(arg(1))) {
+      $northstar_id = arg(2,request_path());
+      $query = db_select('field_data_field_northstar_id', 'f')
```

Comment:
try abstracting this `select` out into a function like `dosomething_user_get_user_by_mobile`